text/0083-func-config-standard.md

```diff
@@ -0,0 +1,66 @@
+- **TEP**: [0](https://github.com/ton-blockchain/TEPs/pull/0)
+- **title**: FunC project configuration file
+- **status**: Draft
+- **type**: Meta
+- **authors**: [Danila Volkov](https://github.com/dvlkv), [Andrey Pfau](https://github.com/andreypfau)
+- **created**: 15.08.2022
+
+# Summary
+
+I propose to make a FunC project config, like project.json, Cargo.toml, etc.
+
+# Motivation
+
+Extensions, compilers and other dev tools need universal project configuration.
+
+# Guide
+
+Extensions can use it for optimal indexing source code, compilers can be configured without bash scripts.  
+Also, here comes more convinient project bootstapping. 
+
+# Specification
+
+I suppose it should be yaml file, named like .func.yml or something else.
+In first implementation it should contain:
+- root source folder
+- entrypoints (contract root files), relative to source folder
```

Comment:
I suggest to add `(optional) metadata` so it'll be easy to publish repos somewhere

```
meta:
    name: "My cool project"
    version: "0.0.1"
    description: "NFT to the moon"
    repository: "uri://my_cool_project"
    homepage: "uri://my_cool_project/docs"
    license: "WTFPL"
    author: "Cool Guy"

    contracts:
      collection:
        name: "Collection base smart contract"
        description: "Just cool collection smart"

      nft:
        name: "Nft base smart contract"
        description: "It's used in collection smart contract in c4, so you need to pass this code to collection"
```

Reply:
Would be better to have an additional standard, there are dozens of things to discuss only about meta.

Reply:
You think so? It seems to me that it breeds entities out of thin air. It feels like a small number of fields is enough.

Reply:
Another possible sense idea would be to describe all the extensions in this md file :)